packages/shared/scope-hoisting/src/formats/commonjs.js

```diff
@@ -441,39 +438,51 @@ export function generateExports(
             rename(path.scope, exportAs, path.scope.generateUid(exportAs));
           }
 
-          let binding = nullthrows(path.scope.getBinding(symbol));
-          if (!hasReplacement) {
-            let id = !t.isValidIdentifier(exportAs)
-              ? path.scope.generateUid(exportAs)
-              : exportAs;
-            // rename only once, avoid having to update `replacements` transitively
-            rename(path.scope, symbol, id);
-            replacements.set(symbol, id);
-            symbol = id;
-          }
+          let binding = path.scope.getBinding(symbol);
+          if (binding) {
+            if (!hasReplacement) {
+              let id = !t.isValidIdentifier(exportAs)
+                ? path.scope.generateUid(exportAs)
+                : exportAs;
+              // rename only once, avoid having to update `replacements` transitively
+              rename(path.scope, symbol, id);
+              replacements.set(symbol, id);
+              symbol = id;
+            }
 
-          let [stmt] = binding.path.getStatementParent().insertAfter(
-            EXPORT_TEMPLATE({
-              NAME: t.identifier(exportAs),
-              IDENTIFIER: t.identifier(symbol),
-            }),
-          );
-          binding.reference(stmt.get<NodePath<Identifier>>('expression.right'));
-
-          // Exports other than the default export are live bindings. Insert an assignment
-          // after each constant violation so this remains true.
-          if (exportAs !== 'default') {
-            for (let path of binding.constantViolations) {
-              let [stmt] = path.insertAfter(
-                EXPORT_TEMPLATE({
-                  NAME: t.identifier(exportAs),
-                  IDENTIFIER: t.identifier(symbol),
-                }),
-              );
-              binding.reference(
-                stmt.get<NodePath<Identifier>>('expression.right'),
-              );
+            let [stmt] = binding.path.getStatementParent().insertAfter(
+              EXPORT_TEMPLATE({
+                NAME: t.identifier(exportAs),
+                IDENTIFIER: t.identifier(symbol),
+              }),
+            );
+            binding.reference(
+              stmt.get<NodePath<Identifier>>('expression.right'),
+            );
+
+            // Exports other than the default export are live bindings. Insert an assignment
+            // after each constant violation so this remains true.
+            if (exportAs !== 'default') {
+              for (let path of binding.constantViolations) {
+                let [stmt] = path.insertAfter(
+                  EXPORT_TEMPLATE({
+                    NAME: t.identifier(exportAs),
+                    IDENTIFIER: t.identifier(symbol),
+                  }),
+                );
+                binding.reference(
+                  stmt.get<NodePath<Identifier>>('expression.right'),
+                );
+              }
             }
+          } else {
```

Comment:
I've added a comment:
```js
            // `getExportedSymbols` can return `$id$import$foo` symbols so that cross-bundle imports
            // are resolved correctly. There is no binding in that case.
```